pkg/supervisor/supervisor.go

```diff
@@ -135,6 +137,13 @@ func (s *Supervisor) Supervise() error {
 		restarts := 0
 		for {
 			s.mutex.Lock()
+
+			if s.CleanBeforeFn != nil {
+				err := s.CleanBeforeFn()
+				if err != nil {
+					s.log.Warnf("Failed to clean before running the process %s: %s", s.BinPath, err)
```

Comment:
I think if there is a cleanup function and if it fails it might have sense to fail supervisor as well

Reply:
It's running in a goroutine, so it'll just stop supervising